lib/rendition.js

```diff
@@ -42,17 +102,158 @@ class Rendition {
     }
 
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
+    }
+
+    async sha1() {
+        if (!this._sha1) {
+            this._sha1 = await fileHash(this.path, HASH_ALGORITHM);
+        }
+        return this._sha1;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    exists() { // see and read the file
+        return fs.existsSync(this.path);
+    }
+
+    async setMimeInformation(mime=null, encoding=null){
+        // for javascript workers mostly ------------------------------------------------------
+        // explicitly set mimetype would get priority
+        if(mime){ // to handle case when this._mime is set to default DEFAULT_CONTENT_TYPE
+            this._mime = mime;
+        }
+
+        // explicitly set encoding would get priority
+        if(encoding){ // to handle case when this.encoding would have a default value
+            this._encoding = encoding;
+        }
+
+        // shellscript mostly ------------------------------------------------------
+        if(this.mimeInfoPath && await fs.pathExists(this.mimeInfoPath)){ // use the path containing the file with mime information
+            let mimeInfoContent = await fs.readFile(this.mimeInfoPath);
+            mimeInfoContent = mimeInfoContent.toString();
+            const mimeInfo = mimeInfoContent.split(";");
+
+            if(mimeInfo && mimeInfo.length === 2){ // we got something in an expected format
+                this._mime = mime || mimeInfo[0].trim() || DEFAULT_CONTENT_TYPE;
+                this._encoding = encoding || mimeInfo[1].trim();
+            }
+        } 
+        // fallback if things are missing ------------------------------------------------------
+        if(!this._encoding || !this._mime){ // fallback to using file-type lib (won't set encoding)
+            try { 
+                await commandExists('file'); // throws if `file` is not found
+                const fileCmdResult = await getMimeInformationFromCommand(this);
+                this._mime = this._mime || fileCmdResult.mime;
+                this._encoding = this._encoding || fileCmdResult.encoding;
+            } catch(err){
+                console.log('`file` command not found', err); // err should be empty
```

Comment:
This can also end up if `file` fails for some reason, so the log message is not always true. You can just change it to `file command failed:` and the logging of the `err` will tell what the problem was, including exit code 127 for not found IIRC.